components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/model/impl/SubscriptionDataStoreImpl.java

```diff
@@ -250,4 +273,64 @@ public void run() {
             }
         }
     }
+
+    public boolean isApisInitialized() {
+
+        return apisInitialized;
+    }
+
+    public void setApisInitialized(boolean apisInitialized) {
+
+        this.apisInitialized = apisInitialized;
+    }
+
+    public boolean isApplicationsInitialized() {
+
+        return applicationsInitialized;
+    }
+
+    public void setApplicationsInitialized(boolean applicationsInitialized) {
+
+        this.applicationsInitialized = applicationsInitialized;
+    }
+
+    public boolean isSubscriptionsInitialized() {
+
+        return subscriptionsInitialized;
+    }
+
+    public void setSubscriptionsInitialized(boolean subscriptionsInitialized) {
+
+        this.subscriptionsInitialized = subscriptionsInitialized;
+    }
+
+    public boolean isApplicationKeysInitialized() {
+
+        return applicationKeysInitialized;
+    }
+
+    public void setApplicationKeysInitialized(boolean applicationKeysInitialized) {
+
+        this.applicationKeysInitialized = applicationKeysInitialized;
+    }
+
+    public boolean isApplicationPoliciesInitialized() {
+
+        return applicationPoliciesInitialized;
+    }
+
+    public void setApplicationPoliciesInitialized(boolean applicationPoliciesInitialized) {
+
+        this.applicationPoliciesInitialized = applicationPoliciesInitialized;
+    }
+
+    public boolean isSubscriptionPoliciesInitialized() {
+
+        return subscriptionPoliciesInitialized;
+    }
+
+    public void setSubscriptionPoliciesInitialized(boolean subscriptionPoliciesInitialized) {
+
+        this.subscriptionPoliciesInitialized = subscriptionPoliciesInitialized;
+    }
```

Comment:
I think we only need getter methods for this case. Also we could add a single method to check whether everything is initialized (result of all the boolean values) instead of having separate getter methods.

Reply:
+1

Reply:
fixed with 41eb399b03e35d34e67d40d8e42ad37f1d3e9dd6